### DISCUSSION TITLE
Tutorial: Adding Starter Page and Template Patterns

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#180] [Topic Ideas]
Currently, there is little documentation for adding Starter Page Patterns and Starter Template Patterns. There are plans to add docs for these in the Theme Handbook: https://github.com/WordPress/Documentation-Issue-Tracker/issues/816 But it will likely be a bit before those are written and published.

In the meantime, it'd be great to introduce them here on the Dev Blog. The Twenty Twenty-Four theme also includes some starter patterns, so it might mean a good chance to show off what the theme is capable of too.

**Starter Page Patterns** are patterns that appear when creating a new page (can also be used for any post type), allowing the user to choose a pattern to start from.

**Starter Template Patterns** are patterns that are available when a user creates a new template of that type in the Site Editor.

It's possible that we could even split this into two separate tutorials.

**Comment by marybaum** [comment]
I like this! I bet there’s also a way to load patterns programmatically based on the existence of data on a given topic.

**Reply by marybaum** [reply]
I mean, I know there is 😜 and it’s even possible to configure a query pattern to account for a couple different conditions. 

One post could be >>>

From no-code to no-hands: three ways to generate posts in a taxonomy term 

On stltennishall.org/category/indux , each of those plaques is an inductee.

We have varying amounts of information on each one, from absolutely zero to Jimmy Connors, where we could pull in all kinds of stuff. Most inductees, dead or alive, will wind up with a video at least. Living inductees also have a profile: What’s up with Maria Steinbach? Etc.

The no-code way is, switch templates By hand , and manually insert the pattern.

In-between is, configure the query loop so if there’s nothing tagged with the inductee, the post just shows a background image and the featured, which is the plaque. 

The no-hands in my world would be a conditional in PHP that could load the templates or just the patterns or whatever. Could even be a switch statement.

**Comment by justintadlock** [comment]
Discussion is moved to the issue after approval. Please continue the conversation there 😃

- https://github.com/WordPress/developer-blog-content/issues/194
